Bluepill-runner/Bluepill-runner/BPRunner.m

```diff
@@ -81,6 +86,129 @@ + (instancetype)BPRunnerWithConfig:(BPConfiguration *)config
     return runner;
 }
 
+- (void)createSimulatorAndInstallAppWithBundles:(NSArray<BPXCTestFile *>*)testBundles {
+    NSString *simulatorUDIDString = nil;
+    if (self.config.appBundlePath) {
+        // This is for integration testing inside Xcode
+        simulatorUDIDString = [self installApplicationwithHost:self.config.appBundlePath withError:nil];
+        [self.testHostForSimUDID setValue:simulatorUDIDString forKey:self.config.appBundlePath];
+        [BPUtils printInfo:INFO withString:@"created sim template: %@ for app host: %@", simulatorUDIDString, self.config.appBundlePath];
+    } else {
+        // This is for testing in command line when we pass the xctestrun file
+        NSMutableSet *hostBundles = [[NSMutableSet alloc] init];
+        for (BPXCTestFile* bundle in testBundles) {
+            [hostBundles addObject:bundle.testHostPath];
+        }
+        if ([testBundles count] == 0) {
+            [BPUtils printInfo:ERROR withString:@"no host bundle founnd!"];
+        }
+        for (NSString *appPath in hostBundles) {
+            NSError *error = nil;
+            simulatorUDIDString = [self installApplicationwithHost:appPath withError:error];
+            [BPUtils printInfo:INFO withString:@"created sim template: %@ for app host: %@", simulatorUDIDString, appPath];
+            [self.testHostForSimUDID setValue:simulatorUDIDString forKey:appPath];
+        }
+    }
+    return;
+}
+
+- (NSString *)installApplicationwithHost:(NSString *)testHost withError:(NSError *)error {
+    SimServiceContext *sc = [SimServiceContext sharedServiceContextForDeveloperDir:self.config.xcodePath error:&error];
+    if (!sc) {
+        [BPUtils printInfo:ERROR withString:[NSString stringWithFormat:@"SimServiceContext failed: %@", [error localizedDescription]]];
+        exit(1);
```

Comment:
Definitively not a big fan of all the `exit(1)`, they will interfere with the exit status we track [here](https://github.com/linkedin/bluepill/blob/3bcef0c9c53cdb3877dfb9365d1dc9cfc4137b36/Bluepill-cli/Bluepill-cli/Bluepill/Reporters/BPExitStatus.m#L17).

Reply:
fixed with deleting simulator when create/install fails and return nil.